docs/topics/compatibility-guide-17.md

```diff
@@ -0,0 +1,502 @@
+> - \>= 1.8.0: remove this property
+
+### Remove kotlin.experimental.coroutines Gradle DSL option and kotlin.coroutines Gradle property
+
+> **Issue**: [KT-50494](https://youtrack.jetbrains.com/issue/KT-50494)
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: remove the `kotlin.experimental.coroutines` Gradle DSL option and the `kotlin.coroutines` property
+>
+> **Deprecation cycle**:
+>
+> - 1.6.20: raise the deprecation level to a warning
+> - 1.7.0: remove the DSL option, its enclosing `experimental` block, and the property
+
+### Deprecate useExperimentalAnnotation compiler option
+
+> **Issue**: [KT-47763](https://youtrack.jetbrains.com/issue/KT-47763)
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: remove the hidden `useExperimentalAnnotation()` Gradle function used to opt in to using an API in a module.
+> `optIn()` function can be used instead
+> 
+> **Deprecation cycle:**
+> 
+> - 1.6.0: hide the deprecation option
+> - 1.7.0: remove the deprecated option
+
+### Deprecate kotlin.compiler.execution.strategy system property
+
+> **Issue**: [KT-51830](https://youtrack.jetbrains.com/issue/KT-51830)
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: deprecate the `kotlin.compiler.execution.strategy` system property used to choose a compiler execution strategy. 
+> `use the Gradle property `kotlin.compiler.execution.strategy` or the compile task property `compilerExecutionStrategy` instead
+>
+> **Deprecation cycle:**
+>
+> - 1.7.0: raise the deprecation level to a warning
+> - \> 1.7.0: remove the property
+
+### Remove kotlinOptions.jdkHome compiler option
+
+> **Issue**: [KT-46541](https://youtrack.jetbrains.com/issue/KT-46541)
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: remove the `kotlinOptions.jdkHome` compiler option used to include a custom JDK from the specified location 
+> into the classpath instead of the default `JAVA_HOME`. Use [Java toolchains](gradle.md#gradle-java-toolchains-support) instead
+>
+> **Deprecation cycle:**
+>
+> - 1.5.30: raise the deprecation level to a warning
+> - \> 1.7.0: remove the option
+
+### Remove noStdlib compiler option
+
+> **Issue**: [KT-49011](https://youtrack.jetbrains.com/issue/KT-49011)
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: remove the `noStdlib` compiler option. Gradle always resolves the Kotlin standard library and adds it to the compiler classpath
+>
+> **Deprecation cycle:**
+>
+> - 1.5.30: raise the deprecation level to a warning
+> - 1.7.0: remove the option
+
+### Remove kotlin2js and kotlin-dce-plugin plugins
+
+> **Issue**: [KT-48276](https://youtrack.jetbrains.com/issue/KT-48276)
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: remove the `kotlin2js` and `kotlin-dce-plugin` plugins
```

Comment:
Probably worth to add what is the proposed replacement?

Reply:
True. Added